research/conditional/utils/layer_manager.py

```diff
@@ -132,6 +136,8 @@ def log(self, verbosity_level):
             return self.log_light()
         elif verbosity_level == 2:
             return self.log_heavy()
+        elif verbosity_level == -1:
```

Comment:
Maybe 3?

Reply:
the current design is that timing info is always recorded when we have nonzero level of verbosity. I agree the design can be improved, but 3 is definitely not the way to go here